pkg/volume/glusterfs/glusterfs.go

```diff
@@ -822,11 +826,14 @@ func (p *glusterfsVolumeProvisioner) CreateVolume(gid int) (r *v1.GlusterfsVolum
 		return nil, 0, "", fmt.Errorf("failed to get cluster nodes for volume %s: %v", volume, err)
 	}
 
-	// The 'endpointname' is created in form of 'glusterfs-dynamic-<claimname>'.
+	// if customEpNamePrefix has not been set, the 'endpointname' is created in form of //'glusterfs-dynamic-<PVC-UID>'. If it has been set, the endpointname is created in form of
+	//<yourprefix>-<PVC-UID>.
 	// createEndpointService() checks for this 'endpoint' existence in PVC's namespace and
 	// If not found, it create an endpoint and service using the IPs we dynamically picked at time
 	// of volume creation.
-	epServiceName := dynamicEpSvcPrefix + p.options.PVC.Name
+
+	epServiceName = p.provisionerConfig.customEpNamePrefix + "-" + string(p.options.PVC.UID)
```

Comment:
If user specifies `customepnameprefix: ""` (i.e. empty), I'd expect that the name of the service would be just the UID, not `-UID`.